### PR TITLE
feature/new-waterfall-stacking

### PR DESCRIPTION
**Pull request by PaulDalek**
Modified existing waterfall stacking feature by adding new types of stacking (normal and overlap).

### CODE REVIEW

File: js/parts/Series.js [hunk 1]
```diff
@@ -960,8 +960,9 @@ null
      * Whether to stack the values of each series on top of each other. Possible
      * values are `undefined` to disable, `"normal"` to stack by value or
      * `"percent"`. When stacking is enabled, data must be sorted in ascending
-     * X order. A special stacking option is with the streamgraph series type,
-     * where the stacking option is set to `"stream"`.
+     * X order. There are also two special stacking options. The first one is
+     * reserved for streamgraph series type, where the stacking option is
+     * set to `"stream"` and `"overlap"` prepared for waterfall series type.
```
Comment (TorsteinHonsi) [inline js/parts/Series.js]: Better sentence:
> ... where the stacking option is set to `"stream"`. The second one is `"overlap"`, which only applies to waterfall series.

File: js/parts/Series.js [hunk 2]
```diff
@@ -960,8 +960,9 @@ null
...
      *
```
Comment (TorsteinHonsi) [inline js/parts/Series.js]: Add `@sample` tags for the two new waterfall stacking samples.

File: js/parts/Series.js [hunk 3]
```diff
@@ -982,6 +982,10 @@ null
      *         Bar
      * @sample {highcharts} highcharts/plotoptions/series-stacking-percent-area/
      *         Area
+     * @sample {highcharts} highcharts/plotoptions/series-stacking-waterfall-normal
+     *         Column
```
Comment (TorsteinHonsi) [inline js/parts/Series.js]: Wrong text. Should be Waterfall with normal stacking

File: js/parts/Series.js [hunk 4]
```diff
@@ -982,6 +982,10 @@ null
...
+     * @sample {highcharts} highcharts/plotoptions/series-stacking-waterfall-overlap
+     *         Column
```
Comment (TorsteinHonsi) [inline js/parts/Series.js]: Wrong text. Should be Waterfall with overlap stacking

File: js/parts/Series.js (hunk 3 above)
Comment (TorsteinHonsi) [reply js/parts/Series.js]: The _name_ still needs to be changed. "Column" is not the correct name.

File: js/parts/Series.js (hunk 4 above)
Comment (TorsteinHonsi) [reply js/parts/Series.js]: The _name_ still needs to be changed. "Column" is not the correct name.

File: js/parts/Series.js (hunk 3 above)
Comment (PaulDalek) [reply js/parts/Series.js]: Ok, sorry for the misunderstanding. Corrected.

**Pull request merged**